Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -1097,7 +1075,180 @@ void IMixedRealityTouchHandler.OnTouchCompleted(HandTrackingInputEventData event
         }
 
         void IMixedRealityTouchHandler.OnTouchUpdated(HandTrackingInputEventData eventData){}
+        #endregion TouchHandlers
+
+        #region InputHandlers
+        public void OnInputUp(InputEventData eventData)
+        {
+            if ((!CanInteract() && !HasPress))
+            {
+                return;
+            }
+
+            if (ShouldListenToUpDownEvent(eventData))
+            {
+                SetPress(false);
+                SetInputUp();
+                if (IsInputFromNearInteraction(eventData))
+                {
+                    // TODO:what if we have two hands grabbing?
+                    SetGrab(false);
+                }
+
+                eventData.Use();
+            }
+            pressingInputSources.Remove(eventData.InputSource);
+        }
+
+        public void OnInputDown(InputEventData eventData)
+        {
+            if (!CanInteract())
+            {
+                return;
+            }
+
+            if (ShouldListenToUpDownEvent(eventData))
+            {
+                pressingInputSources.Add(eventData.InputSource);
+                SetInputDown();
+                SetGrab(IsInputFromNearInteraction(eventData));
+
+                eventData.Use();
+            }
+        }
+
+        /// <summary>
+        /// Public method that can be used to set state of interactable
+        /// corresponding to an input going down (select button, menu button, touch) 
+        /// </summary>
+        public void SetInputDown()
+        {
+            if (!CanInteract())
+            {
+                return;
+            }
+            dragStart3 = null;
+            dragStart2 = null;
```

Comment:
If I read it correctly, this seems to imply that pressing a button in any controller focusing the interactable will reset any drag operation progress. Is that right?

Reply:
For now, yes. Interactable is not really supporting two handed interactions correctly now. I'll leave that for a different PR though.